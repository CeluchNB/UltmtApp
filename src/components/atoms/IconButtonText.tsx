import * as React from 'react'
import { IconButton } from 'react-native-paper'
import { useColors } from '../../hooks'
import {
    StyleProp,
    StyleSheet,
    Text,
    TouchableOpacity,
    ViewStyle,
} from 'react-native'

interface IconButtonTextProps {
    text: string
    icon: string
    onPress: () => void
    style?: StyleProp<ViewStyle>
}

const IconButtonText: React.FC<IconButtonTextProps> = ({
    text,
    icon,
    onPress,
    style,
}) => {
    const { colors } = useColors()

    const styles = StyleSheet.create({
        container: {
            flexDirection: 'column',
        },
        button: {
            padding: 0,
            margin: 0,
            alignSelf: 'center',
        },
        text: {
            padding: 0,
            margin: 0,
            alignSelf: 'center',
            alignContent: 'center',
            color: colors.textPrimary,
        },
    })

    return (
        <TouchableOpacity style={[styles.container, style]} onPress={onPress}>
            <IconButton
                style={styles.button}
                color={colors.textPrimary}
                icon={icon}
                onPress={onPress}
            />
            <Text style={styles.text}>{text}</Text>
        </TouchableOpacity>
    )
}

export default IconButtonText
