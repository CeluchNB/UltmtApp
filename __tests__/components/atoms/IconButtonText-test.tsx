import IconButtonText from '../../../src/components/atoms/IconButtonText'
import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'

it('should match snapshot', () => {
    const snapshot = render(
        <IconButtonText text="text" icon="email-outline" onPress={() => {}} />,
    ).toJSON()
    expect(snapshot).toMatchSnapshot()
})

it('should call on press when pressing on icon', async () => {
    const spy = jest.fn()
    const { getByTestId } = render(
        <IconButtonText text="text" icon="email-outline" onPress={spy} />,
    )
    fireEvent.press(getByTestId('icon-button'))
    expect(spy).toHaveBeenCalled()
})

it('should call on press when pressing on text', async () => {
    const spy = jest.fn()
    const { getByText } = render(
        <IconButtonText text="text" icon="email-outline" onPress={spy} />,
    )
    fireEvent.press(getByText('text'))
    expect(spy).toHaveBeenCalled()
})
