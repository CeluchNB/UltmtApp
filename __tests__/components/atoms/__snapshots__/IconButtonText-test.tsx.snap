// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "flexDirection": "column",
      "opacity": 1,
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#ffd600",
          "fontSize": 25,
        },
        {
          "alignSelf": "center",
          "margin": 0,
          "padding": 0,
        },
        {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
    testID="icon-button"
  >
    ó°‡°
  </Text>
  <Text
    style={
      {
        "color": "#ffd600",
        "margin": 0,
        "padding": 0,
      }
    }
  >
    text
  </Text>
</View>
`;
